### ISSUE TITLE
Launching nyxt with url parameter leads to double loaded buffer

### ISSUE THREAD

**Issue by jellelicht** [atlas-engineer/nyxt#926]
**Describe the bug**
Starting nyxt with a url leads to this url being loaded twice: it seems the url is loaded -> the session is saved -> the url is loaded twice (once as part of "opening nyxt", once by restoring the session that we literally just now saved).

**Precise recipe to reproduce the issue**
On my extremely slow T400 laptop:

init.lisp:
```
(define-configuration browser
  ((session-restore-prompt :always-restore)))
```
... and then: `nyxt "https://google.nl"`
**Information**
- OS name+version: Guix System (master @ `abafdb2aacefef0a2ecf46c8b06b52343727f34f`)
- Graphics card and driver: GM4500, default intel driver
- Desktop environment / Window manager name+version: EXWM 0.24
- How you installed Nyxt (Guix pack, package manager, build from source): guix package (2-pre-release-1)
- Information from command copy-system-information:

```
Nyxt version: 2-pre-release-1 
Operating system kernel: Linux 4.19.142-gnu
Lisp implementation: SBCL 2.0.8
Features: (WEBKIT2 WEBKIT2-2.28 WEBKIT2-EMOJI WEBKIT2-MEDIA WEBKIT2-SANDBOXING
           GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16 GTK-3-14 GTK-3-12 GTK-3-10
           GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22 GDK-3-20 GDK-3-18 GDK-3-16
           GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8 GDK-3-6 GDK-3-4 CAIRO-1-10
           CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32 GLIB-2-34 GLIB-2-36
           GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44 GLIB-2-46 GLIB-2-48
           GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56 GLIB-2-58 GLIB
           FSET-EXT-STRINGS SWANK PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES
           PARENSCRIPT SBCL+SAFE-STANDARD-READTABLE NAMED-READTABLES
           OSICAT-FD-STREAMS LPARALLEL 21BIT-CHARS CL-FAD CHUNGA FLEXI-STREAMS
           CLOSER-MOP CL-PPCRE-UNICODE CL-UNICODE CL-PPCRE
           CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
           BORDEAUX-THREADS LPARALLEL.WITH-CLTL2 LPARALLEL.WITH-CAS
           LPARALLEL.WITH-STEALING-SCHEDULER SPLIT-SEQUENCE GRAY-STREAMS
           FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE SEQUENCE-EMPTYP
           FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE CL-JSON-CLOS CL-JSON
           THREAD-SUPPORT ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF OS-UNIX
           NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC 64-BIT ANSI-CL
           COMMON-LISP ELF IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-CORE-COMPRESSION SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)
```

**Output when started from a shell**
```
~/.config/nyxt $ nyxt "google.nl"
Nyxt version 2-pre-release-1
<INFO> [22:50:51] Loading Lisp file "/home/jelle/.config/nyxt/init.lisp".
<INFO> [22:50:51] Using data profile "default".
<INFO> [22:50:51] Listening to socket "/home/jelle/.local/share/nyxt/nyxt.socket".
<INFO> [22:50:51] Loading 2 bookmarks from "/home/jelle/.local/share/nyxt/bookmarks.lisp".
<INFO> [22:50:51] Loading global history of 216 URLs from "/home/jelle/.local/share/nyxt/history.lisp".
<INFO> [22:50:51] Saving session to "/home/jelle/.local/share/nyxt/sessions/default.lisp".
<INFO> [22:50:52] Restoring session "/home/jelle/.local/share/nyxt/sessions/default.lisp".
<INFO> [22:50:52] Restoring (https://google.nl/).
<INFO> [22:50:52] Loading "https://google.nl/".
<INFO> [22:50:52] Saving session to "/home/jelle/.local/share/nyxt/sessions/default.lisp".
<INFO> [22:50:52] Saving session to "/home/jelle/.local/share/nyxt/sessions/default.lisp".
<INFO> [22:50:55] Finished loading "https://www.google.nl/".
<INFO> [22:51:16] Saving session to "/home/jelle/.local/share/nyxt/sessions/default.lisp".
<INFO> [22:51:16] Saving session to "/home/jelle/.local/share/nyxt/sessions/default.lisp".
```

**Comment by Ambrevar** [comment]
Should be fixed with b709b773fc4f65be6e584764b6ad2e9294c24ba0 if
session-restore-prompt is _not_ `:always-ask`.

To fix this properly, we need to make minibuffer queries synchronous.
I'll do this very soon.

Can you confirm it works for you?

**Comment by jellelicht** [comment]
I can confirm it works as I expect it should with the configuration I
reported. Thanks!

**Comment by Ambrevar** [comment]
Synchronous minibuffer is coming: https://github.com/atlas-engineer/nyxt/pull/928
Closing this since it will be fully fixed once #928 is in.
